node_modules/oae-core/userdetails/js/userdetails.js

```diff
@@ -0,0 +1,161 @@
+                };
+                oae.api.user.updateUser(params, function (err, data) {
+                    // If the update failed, enable the form and show an error notification
+                    if (err) {
+                        oae.api.util.notification(
+                            oae.api.i18n.translate('__MSG__PROFILE_NOT_EDITED__'),
+                            oae.api.i18n.translate('__MSG__PROFILE_DETAILS_EDIT_FAIL__', 'userdetails'),
+                            'error');
+
+                        // Enable the form
+                        $('#userdetails-form *', $rootel).prop('disabled', false);
+                        return;
+                    }
+
+                    // If the update succeeded, trigger an event,
+                    // show a success notification and close the modal
+                    $('#userdetails-modal', $rootel).modal('unlock');
+                    $('#userdetails-modal', $rootel).modal('hide');
+                    oae.api.util.notification(
+                        oae.api.i18n.translate('__MSG__PROFILE_EDITED__'),
+                        oae.api.i18n.translate('__MSG__PROFILE_DETAILS_EDIT_SUCCESS__', 'userdetails'));
+
+                    // Trigger an event so other elements can be updated
+                    $(document).trigger('oae.editprofile.done', data);
+                });
+            });
+
+            // Avoid default form submit behaviour
+            return false;
+        };
+
+        /**
+         * Hide the user details form and show the Terms and Conditions
+         */
+        var showTermsAndConditions = function() {
+            $('#userdetails-form', $rootel).hide();
+            $('#userdetails-terms-and-conditions-container', $rootel).show();
+            $('.modal-dialog', $rootel).addClass('termsandconditions');
+            $('.modal-footer', $rootel).show();
+        };
+
+        /**
+         * Show the user details form and hide the Terms and Conditions
+         */
+        var showUserDetailsForm = function() {
+            $('#userdetails-form', $rootel).show();
+            $('#userdetails-terms-and-conditions-container', $rootel).hide();
+            $('.modal-dialog', $rootel).removeClass('termsandconditions');
+            $('.modal-footer', $rootel).hide();
+        };
+
+        /**
+         * Add the bindings to show or hide the Terms and Conditions
+         */
+        var addTermsAndConditionsBinding = function() {
+            // Bind the Terms and Conditions link
+            $rootel.on('click', '#register-terms-and-conditions', function() {
+                // Get the Terms and Conditions and render them if we hadn't done so already
+                if ($('#userdetails-terms-and-conditions-container', $rootel).is(':empty')) {
+                    oae.api.user.getTC(function(err, data) {
+                        oae.api.util.template().render($('#userdetails-termsandconditions-template', $rootel), {
```

Comment:
I think this would have to be `userdetails-terms-and-conditions-container-template` to be in line with UI patterns